modules/collect/src/main/java/com/opengamma/strata/collect/io/ArrayByteSource.java

```diff
@@ -175,6 +175,8 @@ public static ArrayByteSource from(ByteSource other) {
       String path = str.substring(23, pos);
       int lastSlash = path.lastIndexOf('/');
       fileName = path.substring(lastSlash + 1);
+    } else if (other instanceof BeanByteSource) {
```

Comment:
I meant stick this first, since you don't need to call `toString()` if this is true, and nest the rest under an else.